athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandler.java

```diff
@@ -261,6 +267,15 @@ public ListTablesResponse doListTables(BlockAllocator blockAllocator, ListTables
 
     /**
      * Returns a list of tables from AWS Glue DataCatalog with optional filtering for the requested schema (aka database)
+     * The request (ListTablesRequest) includes a nextToken (String) and pageSize (int) attributes that can
+     * be used to paginate the response. The nextToken (request.getNextToken()) will be null if this is the first
+     * paginated request, or will contain a value of the starting point for the current request. pageSize (request
+     * .getPageSize()) will be -1 (ListTablesRequest.UNLIMITED_PAGE_SIZE_VALUE) for non-paginated requests (full
+     * list of tables should be returned), or it will be an integer value > 0. The latter indicates the maximum
+     * number of tables to return in the response (ListTablesResponse).
+     * The response (ListTablesResponse) also has a nextToken (String) attribute used for pagination. Its value
+     * should be set with the starting point for the next paginated request. If all tables have been processed, the
+     * value returned for nextToken should be null.
```

Comment:
same comment as before about copy pasting large blobs of documentation

Reply:
Reduced